src/libraries/System.Private.CoreLib/src/System/Threading/Thread.OSX.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal static class ThreadOSX
+    {
+        private static bool CheckEnableAutoreleasePool()
+        {
+            const string feature = "System.Threading.Thread.EnableAutoreleasePool";
+#if !CORECLR
+            return AppContextConfigHelper.GetBooleanConfig(feature, false);
+#else
+            bool isEnabled = CLRConfig.GetBoolValue(feature, out bool isSet);
+            if (!isSet)
+                return false;
+
+            return isEnabled;
+#endif
+        }
+
+        public static bool EnableAutoreleasePool { get; } = CheckEnableAutoreleasePool();
+    }
+
+    public sealed partial class Thread
+    {
+        [UnmanagedCallersOnly]
+        private static void CallDrain(IntPtr p)
```

Comment:
This also got pretty CoreCLR specific. I would move it under coreclr/src. For other runtimes that have more of the threading refactored to managed code, it would be a regular managed call (e.g. from here - https://github.com/dotnet/runtime/blob/65bcecebde4aebd5926952ccfc4dc8e7fbf7eaed/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs#L305). No unmanaged pointers involved.

Reply:
My intent here is to have Mono call this same API though for adding pools to managed threads.

Reply:
I moved the API and will add something simpler to Mono. The exit point above makes sense and will hopefully make the Mono changes simpler.